bundles/AdminBundle/Event/ManyToManyRelationAssetUploadSubscriber.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * Pimcore
+ *
+ * This source file is available under two different licenses:
+ * - GNU General Public License version 3 (GPLv3)
+ * - Pimcore Commercial License (PCL)
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ *  @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ *  @license    http://www.pimcore.org/license     GPLv3 and PCL
+ */
+
+namespace Pimcore\Bundle\AdminBundle\Event;
+
+use Pimcore\Bundle\AdminBundle\Helper\AssetTypeHelper;
+use Pimcore\Bundle\AdminBundle\Helper\ManyToManyRelationValidator;
+use Pimcore\Event\AssetEvents;
+use Pimcore\Event\Model\Asset\ResolveUploadTargetEvent;
+use Pimcore\Model\DataObject\ClassDefinition\Data\ManyToManyRelation;
+use Pimcore\Model\DataObject\Concrete;
+use Pimcore\Model\Element\ValidationException;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+final class ManyToManyRelationAssetUploadSubscriber implements EventSubscriberInterface
```

Comment:
Looks like you forgot to commit the `service.yml` file where you added the event listener as a service.

Furthermore, I would move the file into the `EventListener` folder and rename it according to the other Symfony naming conventions (so it would be `ManyToManyRelationAssetUploadListener.php`)

Reply:
I used the Symfony naming convention @see https://symfony.com/doc/current/event_dispatcher.html or https://symfony.com/doc/current/components/event_dispatcher.html#using-event-subscribers

For EventSubscribers we don't need the service.yml (only implementing the EventSubscriberInterface).

In my opinion we should rename the EventListener folder to EventSubscriber or Event. All classes implement the EventSubscriberInterface interface (but not part of this PR).